src/host/host.cpp

```diff
@@ -79,15 +84,23 @@ int main(int argc, char *argv[]) {
     RPCHandler::setEnclaveArgs(argc, argv);
 
     try {
+        std::shared_ptr<ThreadFactory> threadFactory =
+            std::shared_ptr<ThreadFactory>(new ThreadFactory());
+        std::shared_ptr<ThreadManager> threadManager =
+            ThreadManager::newSimpleThreadManager(4);
+        threadManager->threadFactory(threadFactory);
+        threadManager->start();
+
         auto handler = std::make_shared<RPCHandler>();
         auto processor = std::make_shared<RPCProcessor>(handler);
         auto serverTransport = std::make_shared<TServerSocket>(HOST_PORT);
         auto transportFactory = std::make_shared<TBufferedTransportFactory>();
         auto protocolFactory = std::make_shared<TBinaryProtocolFactory>();
 
         std::shared_ptr<apache::thrift::server::TServer> server;
-        server.reset(new TThreadedServer(processor, serverTransport,
-                                         transportFactory, protocolFactory));
+        server.reset(new TThreadPoolServer(processor, serverTransport,
+                                           transportFactory, protocolFactory,
+                                           threadManager));
```

Comment:
ThreadPoolServer to limit the number of threads that are calling into the enclave